.github/workflows/ci.yaml

```diff
@@ -4,6 +4,8 @@ on: # yamllint disable-line rule:truthy
   pull_request:
     paths:
       - "server/**"
+      - "hack/tools/**"
```

Comment:
we may want to test if hack/tools are updated. As an example, kustomize could introduce breaking changes in manifests (like kustomize v4 to v5). WDYT?

Reply:
maybe kustomize is not the best example, as this is for linting and unit tests. Breaking changes to mockgen is a better example

Reply:
I'm not sure I understand; this line checks if anything underneath `hack/tools/` is a part of the diff (and triggers this workflow if so).  What would be different?

Reply:
oh sorry, I thought it was the exclusion list, not the inclusions 🤦🏾

then, should we rename this file to `ci-server.yaml` and add similar ones for `operator` and `e2e`?

Reply:
that's a `TODO` still for me, I'll do that in a follow-up PR once I have `operator` CI better defined.